### ISSUE TITLE
Keeping last layer names in Stacked Model

### ISSUE THREAD

**Issue by TkinterinShanghai** [tensorflow/tensorflow#61923]
### Issue type

Feature Request

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

2.13.0

### Custom code

No

### OS platform and distribution

macOS 13.4

### Python version

3.9

### Current behavior?

When using Model Stacking, I would like to access the last layer in the last stacked model to pass individual loss functions to them
```python
# ------------------ Encoder ------------------
encoder_model = Model(inputs=all_inputs, outputs=latent_space, name="encoder")
# ... irrelevant code 

# ------------------ Decoder ------------------
# Output Layers
numeric_output = Dense(
    self.numeric_dim, activation="linear", name="numeric_output")(decoder2)
binary_output = Dense(
    self.binary_dim, activation="sigmoid", name="binary_output")(decoder2)

decoder_output = [numeric_output] + [binary_output]
decoder_model = Model(inputs=latent_input, outputs=decoder_output, name="decoder")

# ------------------ Autoencoder ------------------
autoencoder_output = decoder_model(encoder_model(all_inputs))
autoencoder = Model(inputs=all_inputs, outputs=pass_through_layers, name="autoencoder")

# This will not work:
losses = {
            "numeric_output": "mse",
            "binary_output": "binary_crossentropy"
}
autoencoder.compile(optimizer=Adam(learning_rate=lr), loss=losses)
```

The last layer will be renamed to decoder1, decoder2, etc. 
It would be much nicer if there was some passthrough argument in a model that would allow referencing the last layer in the stacked model directly

### Standalone code to reproduce the issue

```shell
This will fail


import numpy as np
from tensorflow.keras.layers import Input, Dense
from tensorflow.keras.models import Model
from tensorflow.keras.optimizers import Adam

# Input dimensions
input_dim = 10
latent_dim = 5
numeric_dim = 3
binary_dim = 2

# Learning rate
lr = 0.001

# Number of samples
n_samples = 1000

# Generate random data
X = np.random.rand(n_samples, input_dim)
y_numeric = np.random.rand(n_samples, numeric_dim)
y_binary = np.random.randint(0, 2, size=(n_samples, binary_dim))

# ------------------ Encoder ------------------
all_inputs = Input(shape=(input_dim,), name="all_inputs")
latent_space = Dense(latent_dim, activation="relu", name="latent_space")(all_inputs)
encoder_model = Model(inputs=all_inputs, outputs=latent_space, name="encoder")

# ------------------ Decoder ------------------
latent_input = Input(shape=(latent_dim,), name="latent_input")
decoder2 = Dense(10, activation="relu", name="decoder2")(latent_input)

numeric_output = Dense(numeric_dim, activation="linear", name="numeric_output")(decoder2)
binary_output = Dense(binary_dim, activation="sigmoid", name="binary_output")(decoder2)

decoder_output = [numeric_output, binary_output]
decoder_model = Model(inputs=latent_input, outputs=decoder_output, name="decoder")

# ------------------ Autoencoder ------------------
autoencoder_output = decoder_model(encoder_model(all_inputs))

# This will not work:
losses = {
    "numeric_output": "mse",
    "binary_output": "binary_crossentropy"
}
autoencoder = Model(inputs=all_inputs, outputs=autoencoder_output, name="autoencoder")
autoencoder.compile(optimizer=Adam(learning_rate=lr), loss=losses)

# Fit the model
autoencoder.fit(
    x=X,
    y=[y_numeric, y_binary],
    epochs=10,
    batch_size=32
)
```

**Comment by sachinprasadhs** [comment]
@TkinterinShanghai , Thanks for reporting the issue.

Since this feature request is specific to Keras, could you please close this and open a new issue in Keras repo https://github.com/keras-team/keras/issues
